plugins/pulp_rpm/plugins/importers/yum/associate.py

```diff
@@ -461,7 +461,7 @@ def _associate_unit(dest_repo, unit, config):
     elif isinstance(unit, models.RPM):
         # copy will happen in one batch
         return unit
-    elif isinstance(unit, (models.YumMetadataFile, models.ModulemdDefaults)):
+    elif isinstance(unit, models.YumMetadataFile):
```

Comment:
Expand the code of the ```associate_copy_for_repo()``` function below.  With ```set_content``` as True, it sets the storage path on the new unit being created.

However, I was getting failures because ```unit._storage_path``` was ```None```.

I assume that this is not the appropriate place to fix this, however I'm not sure what the actual bug is.  Is the problem that ```unit._storage_path``` is ```None```?  Should it always be set to something since ```models.ModulemdDefaults``` inherits from ```FileContentUnit```?  Is it invalid for this not to be set?

Or, is that valid, and we we just change line 499 to add a ```None```-check?

```if set_content and unit._storage_path:```?

Reply:
I agree this is not the appropriate fix. The reason is that module_defaults should be copied and new unit should be created to avoid situations like this  https://github.com/pulp/pulp_rpm/pull/1146#discussion_r205494063 - it's actually a PR which added module-defaults here.

I'm not sure why _strorage_path is not set, I'd expect it to be set since we are copying an existing unit.
@ipanova , any thoughts?

@dalley, tbh, I'm hesitant to just add a check for None before understanding where the root cause of the problem is.

Reply:
storage path should not be none, i think there might be a problem somewhere else.